pkg/scheduler/framework/score.go

```diff
@@ -46,8 +66,16 @@ type ScoredClusters []*ScoredCluster
 // Len returns the length of a ScoredClusters; it implemented sort.Interface.Len().
 func (sc ScoredClusters) Len() int { return len(sc) }
 
-// Less returns true if a ScoredCluster is of a lower score than another; it implemented sort.Interface.Less().
+// Less returns true if a ScoredCluster is of a lower score than another; when two clusters have
+// the same score, the one with a name that is lexicographically smaller is considered to be the
+// smaller one.
+//
+// It implemented sort.Interface.Less().
 func (sc ScoredClusters) Less(i, j int) bool {
+	if sc[i].Score.Equal(sc[j].Score) {
```

Comment:
nil is a tricky one. What would it mean if both scores are nil? And what would happen if both cluster pointers are nil? In these cases, panic might be a better choice.

Reply:
It's just my personal preference, but maybe the documentation of nil is invalid + checks on the caller level is better as the semanics on nil for this specific method is very much unclear.

Reply:
this util can treat the nil as the smallest, though it has no meaning.

i feel it's better to do the nil check, add the unexpectedBehavior log and ignore it instead of causing the panic.

Reply:
And I recall there are a number of discussions in the Go community promoting not doing the nil check inside the method. Let me do a quick search.

Reply:
https://dave.cheney.net/practical-go/presentations/gophercon-singapore-2019.html#_never_use_nil_to_indicate_failure
https://www.reddit.com/r/golang/comments/tja8cj/do_you_test_that_a_method_receiver_is_nil/

Reply:
Discussed offline; will use a caller beware pattern.